Tools/WinMLRunner/src/BindingUtilities.h

```diff
@@ -367,11 +542,17 @@ namespace BindingUtilities
             throw;
         }
 
+        ITensor tensor;
+        SoftwareBitmap softwareBitmap(nullptr);
         std::vector<std::string> elementStrings;
         if (!args.CsvPath().empty())
         {
             elementStrings = ParseCSVElementStrings(args.CsvPath());
         }
+        else if (!args.ImagePath().empty())
+        {
+            softwareBitmap = LoadImageFile(tensorDescriptor, inputDataType, args.ImagePath().c_str(), args, iterationNum);
```

Comment:
If inputDataType is Tensor then LoadImageFile will fail. There needs a check in CommandLineArgs to make sure that -inputImagePreprocess flag doesn't get passed in with -Tensor.

Reply:
-Tensor itself will only work with an input of csv or garbage data, right?  Images are not valid with that option?  Interestingly enough, the way we're testing -inputImagePreprocess right now with 1 0 0 0, is actually generating a tensor and passing that to WinML… is that a capability that can be leveraged to basically do OnClient Tensorization?  Is this interesting at all?

Reply:
Good point! -Tensor itself will only work with an input of csv or garbage data. Adding a check to make sure that -Tensor only works with csv or garbage data inputs would protect -InputImagePreprocess.

Are you proposing to rename -InputImagePreprocess to something like -OnClientTensorize ?

Reply:
I like it... no problem changing to OnClientTensorize.  Is your intention that we would implement both, or just the one?

Reply:
The usage is inconsistent with the README:  \<msdR\> \<msdG\> \<msdB\> is  \<rFactor\> \<gFactor\> \<bFactor\>.

My intention is to keep your implementation but to just make \<scale\> \<rFactor\> \<gFactor\> \<bFactor\> optional parameters with -OnClientTensorize.

For example, if the user just specifies -OnClientTensorize, then scale = 1, rFactor = 0, gFactor = 0, bFactor = 0